packages/guilded.js/lib/managers/global/MemberBanManager.ts

```diff
@@ -0,0 +1,45 @@
+import Collection from "@discordjs/collection";
+import { MemberBan } from "../../structures/Member";
+import CacheableStructManager from "./CacheableStructManager";
+
+export default class GlobalMemberBanManager extends CacheableStructManager<string, MemberBan> {
+    /** Fetch a member ban in a server */
+    fetch(serverId: string, userId: string) {
+        return this.client.rest.router.getMemberBan(serverId, userId).then(data => {
+            const newMemberBan = new MemberBan(this.client, { ...data.serverMemberBan, serverId })
+            if(this.client.options.cache?.cacheMemberBans !== false) this.client.bans.cache.set(newMemberBan.id, newMemberBan);
+            return newMemberBan;
+        })
+    }
+
+    /** Fetch all bans in a server */
+    fetchMany(serverId: string) {
+        return this.client.rest.router.getMemberBans(serverId).then(data => {
+            const newMemberBans = new Collection<string, MemberBan>();
+            for(const ban of data.serverMemberBans) {
+                const newMemberBan = new MemberBan(this.client, { serverId, ...ban });
+                newMemberBans.set(newMemberBan.id, newMemberBan)
+                if(this.client.options.cache?.cacheMemberBans !== false) this.client.bans.cache.set(newMemberBan.id, newMemberBan);
```

Comment:
We aren't actually caching members here. Nothing in the member ban gives us any member data (there is minimal user data). Though, you make a good point with whether it's useful or not. My thought process was that since bans cannot be edited, only deleted and recreated, that you'd never actually need to make another request for that same ban as it'll never change. However, this brings in the issue about a stale cache, where if the ban does get deleted, we have no way of knowing and the cache would therefore be unreliable. I'll think this over.